### DISCUSSION TITLE
How to properly remove system tray icon before exit?

### DISCUSSION THREAD

**Discussion by boan-anbo** [tauri-apps/tauri#4668] [Q&A]
I'm currently calling `exit` from `@tauri-apps/api/process` to exit the program. The system tray icon, however, is not destroyed after exit. 

How do I make sure the icon is removed from the system tray before the program exists?

**Comment by minosss** [comment]
tested, it works well.

confirm the app process has been killed? or more details

**Comment by boan-anbo** [comment]
Thanks for the reply! Yes, the process manage shows the process has been killed. I'm positing a gif screen recording. The icon remains on the system tray UNTIL the cursor hovers on it---the remaining icon will disappeare.

If I do multiple launching and exiting, it will leave multiple icons on the system tray.

![Animation](https://user-images.githubusercontent.com/7269717/179077240-41a0f993-f505-4c5f-9c5c-47379efa2fc7.gif)

System is Windows. 
Tauri Version 1.0.3
Exit logic:


```

import {exit} from "@tauri-apps/api/process";

async exitPaperTimer() {

    await exit()
}


```

**Reply by FabianLars** [reply]
For whatever reason the js `exit` function calls the wrong `exit` function in the backend, which doesn't clean up the tray.
As a workaround you can add a custom rust command that looks like this:
```rs
#[tauri::command]
fn exit_app(handle: tauri::AppHandle) {
    handle.exit(0);
}
```
See here on how to integrate it into your app: https://tauri.app/v1/guides/features/command

**Reply by boan-anbo** [reply]
Awesome! It works now. Thanks so much!
